### PR TITLE
Add initial conformance results for Contour

### PR DESCRIPTION
**Pull request by sunjayBhatia**
**What type of PR is this?**

/area conformance

**What this PR does / why we need it**:

Adds initial conformance results for Contour's Gateway API implementation

**Which issue(s) this PR fixes**:

N/A

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/results/contour.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      Failed: 0
+      Passed: 28
+      Skipped: 0
+    summary: ""
+  extended:
+    result: success
+    statistics:
+      Failed: 0
+      Passed: 9
+      Skipped: 0
```
Comment (sunjayBhatia) [inline conformance/results/contour.yaml:25]: due to https://github.com/kubernetes-sigs/gateway-api/issues/2206 this is not accurate

File: conformance/results/contour.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2023-07-25T22:11:56Z"
+gatewayAPIVersion: TODO
+implementation:
+  contact:
+  - '@projectcontour/maintainers'
+  organization: projectcontour
+  project: contour
+  url: https://github.com/projectcontour/contour
+  version: "f6c112e60a40ceb9d448354473a8fe7d1a726790"
```
Comment (sunjayBhatia) [inline conformance/results/contour.yaml:10]: current Contour `main` branch commit

File: conformance/results/contour.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2023-07-25T22:11:56Z"
+gatewayAPIVersion: TODO
```
Comment (shaneutt) [inline conformance/results/contour.yaml:3]: We have #2197 to automate adding the `gatewayAPIVersion`, for now just manually enter it

(I assume it's `v0.7.1`?)

```suggestion
gatewayAPIVersion: v0.7.1
```

**Review by shaneutt — COMMENTED** [review]
Awesome, thanks for filing a report! Just need to identify which Gateway API version this report is for and then LGTM.

**Review by Xunzhuo — APPROVED** [review]
LGTM, other than `TODO` gwapi version.

File: conformance/results/contour.yaml (hunk 3 above)
Comment (mlavacca) [reply conformance/results/contour.yaml:3]: +1

Comment (sunjayBhatia) [reply conformance/results/contour.yaml]: I used a commit off of main for the tests and CRDs+webhook manifest

**Review by shaneutt — APPROVED** [review]
/approve
/lgtm

Thanks for reporting!

**Pull request merged**